From 12fe594406b79ac71a66c375115ec00619f2fbdb Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 31 Mar 2017 06:44:15 +0800
Subject: [PATCH 1422/5242] MLK-14585-2 clk: imx: correct pm API for getting
 power domain structure

commit  fdd4ef244998c9db01ef23f8d46cf29010658b40 from
https://source.codeaurora.org/external/imx/linux-imx.git

Should use correct power domain API for getting power
domain structure by phandle, adding a power domain here is
incorrect, replace it with correct API.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-gate-scu.c |    3 +--
 drivers/clk/imx/clk-mux-scu.c  |    2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/imx/clk-gate-scu.c b/drivers/clk/imx/clk-gate-scu.c
index 815e4ce..f70cc0b 100644
--- a/drivers/clk/imx/clk-gate-scu.c
+++ b/drivers/clk/imx/clk-gate-scu.c
@@ -197,9 +197,8 @@ static void populate_gate_pd(struct clk_gate2_scu *clk)
 	if (np) {
 		pd_args.np = np;
 		pd_args.args_count = 0;
-		of_genpd_add_provider_simple(np, clk->pd);
+		clk->pd = genpd_get_from_provider(&pd_args);
 	}
-	return;
 }
 
 /* Write to the LPCG bits. */
diff --git a/drivers/clk/imx/clk-mux-scu.c b/drivers/clk/imx/clk-mux-scu.c
index 968db33..65cdbfd 100644
--- a/drivers/clk/imx/clk-mux-scu.c
+++ b/drivers/clk/imx/clk-mux-scu.c
@@ -68,7 +68,7 @@ static void populate_mux_pd(struct clk_mux_scu *clk)
 	if (np) {
 		pd_args.np = np;
 		pd_args.args_count = 0;
-		of_genpd_add_provider_simple(np, clk->pd);
+		clk->pd = genpd_get_from_provider(&pd_args);
 	}
 }
 
-- 
1.7.9.5

